### ISSUE TITLE
Are non-multiplexed ts (and aac) streams supported?

### ISSUE THREAD

**Issue by architsinha** [videojs/mux.js#137]
Hi,

I am using _mux.js_ as a part of _video-contrib-hls.js_. I was facing an issue with the following stream:
http://demo.unified-streaming.com/video/tears-of-steel/tears-of-steel-sample-aes.ism/.m3u8

I get the exception "_Cannot read property segments of undefined_" . This happens in the function "_processPendingSegments_()_" in _video-contrib-media-sources_. 

It seems to be due to the fact that the mux.js is not returning proper data to it,  the cause for the specific exception is that the the segment.type is returned as combined, where as it expects it to be audio or video.

I have observed this issue with one more stream which is also non-multiplexed. I couldn't find more non-multiplexed HLS streams apart from these two to confirm it further. Also I haven't observed this with any of the multiplexed streams.

Is this a known limitation/expected behavior of the module? Has anyone else observed this issue or am I missing something here?

Versions used:
mux.js - 3.0.4
video-contrib-media-sources - 4.1.2
video-contrib-hls - 4.0.2
video.js - 5.11.9

Browser used: Google Chrome version 55.0.2883.87 m

Regards,
Archit

**Comment by dmlap** [comment]
Two thoughts:
- contrib-hls doesn't support sample-AES encryption right now, only full segment encryption. I'm guessing this is the issue.
- I scanned your m3u8 quickly and I did not see URIs for any of the audio groups, which would indicate your segments have both audio and video tracks (that is, "combined" in mux.js terminology). Do you mean something different than that when you call this stream "non-multiplexed"?

**Comment by architsinha** [comment]
Thanks for the response. Some of my initial analysis was incorrect. I debugged a little further and found the following:

1) By "non-multiplexed", I refer to the case where the audio and video tracks come in different segments However, my initial inference regarding the stream was incorrect. This stream is "multiplexed", that is, it 
has audio and video tracks in the same segment, as you rightly pointed out.

2) Also, as you correctly pointed out, the reason that this stream does play is that this stream has sample AES and currently only full segment AES is supported. I am the guessing the support for sample AES, whenever it is added, will be present in the mux.js module, since it is the one which has access to the samples. Is that planned anytime soon?

3)Regarding the original issue I posted, as I had mentioned, I had observed it in two streams, one of them was what I had shared earlier, the other I am not sure if I can share. The other stream is actually non-multiplexed. I was getting the same error with that stream, however as I later found out, the cause of the errors are different.

4) The issue with the other stream was the following:

a) It has video (only) tracks contained in ts and two audio tracks in aac format.
b) The aac segments did not have ID3 meta data. I saw that the function to detect AAC packets _isLikelyAacData()_, was actually checking for ID3, this was causing the checks to fail and leading to the issue
c) I changed that and some other code, but even then the stream was not playing. I figured out, it was because aac does not have any mechanism to pass timestamp info which is needed to calculate _baseMediaDecodeTime_ , ID3 if present can give that info. So probably that's why the function to check AAC packets was actually checking for ID3. Is my understanding correct?
d) So I guess, this is not really an issue with the module.  The only way to play the stream is to count the samples and derive _baseMediaDecodeTime_ from it, but even then seek would not work.
